### PR TITLE
Add ga tracking for takeover impressions

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Adds ga tracking for takeover impressions

## QA

- Load the homepage (make sure a takeover loads) and check the ga event has been pushed with the appropriate data
Two ways to do this:
1. run the branch locally follow the guide in the [google analytic docs](https://developers.google.com/analytics/devguides/collection/analyticsjs/events) to see tga events in the network tab
2. (Recommended) Download a broswer extension that tracks GA events, I am using [tag legacy](https://chrome.google.com/webstore/detail/tag-assistant-legacy-by-g/kejbdjndbnbjgmefkgdddjlbokphdefk)

Should look something like: 
```
{
"gtm.start" : 1678275907959 ,
"event" : "gtm.js" ,
"gtm.uniqueEventId" : 1
},
{
"event" : "gtm.dom" ,
"gtm.uniqueEventId" : 2
},
{
"event" : "NonInteractiveGAEvent" ,
"eventCategory" : "www.ubuntu.com-impression-takeover" ,
"eventAction" : "from:http://0.0.0.0:8010/ to:/engage/top-five-iot-challenges-cn?utm_medium=takeover" ,
"eventLabel" : "立即下载" ,
"gtm.uniqueEventId" : 3
},
{
"event" : "gtm.load" ,
"gtm.uniqueEventId" : 4
}
```

## Issue / Card

https://warthogs.atlassian.net/browse/WD-2462

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -1497,6 +1497,14 @@ <h2>
       secondaryUrl.href = selectedTakeover.secondary_url;
       secondaryUrl.innerHTML = selectedTakeover.secondary_cta + "&nbsp;&rsaquo;";
     }
+
+    dataLayer.push({
+      event: "NonInteractiveGAEvent",
+      eventCategory: "www.ubuntu.com-impression-takeover",
```
Comment (anthonydillon) [inline templates/index.html:1503]: This is `cn` not `www.` but maybe better categorised as just a "Chinese takeover impression"

File: templates/index.html [hunk 2]
```diff
@@ -1497,6 +1497,14 @@ <h2>
...
+      eventAction: "from:" + window.location.href + " to:" + selectedTakeover.primary_url,
```
Comment (anthonydillon) [inline templates/index.html:1504]: From and to is probably not as useful as the takeover title. We know its a click on the advert. WDYT?

File: templates/index.html [hunk 3]
```diff
@@ -1497,6 +1497,14 @@ <h2>
...
+      eventLabel: selectedTakeover.primary_cta,
```
Comment (anthonydillon) [inline templates/index.html:1505]: Maybe this one should just the takeover title and the eventAction above be simply "CTA click"?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Some questions in the code.

File: templates/index.html (hunk 3 above)
Comment (petesfrench) [reply templates/index.html:1505]: Makes sense to me, will update now

**Review by anthonydillon — APPROVED** [review]
Nice thanks, if we are consistent with the event type the data will become more useful 👍

**Pull request merged**